### PR TITLE
Add CODEOWNERS

### PR DESCRIPTION
**Pull request by grandchild**
Initial autogenerated list from git history by walking tree and:
- Stop descent if directory that contains "Chart.yaml".
- Run "git shortlog -sne" to collect authors.
- Order authors by number of total commits to that path.
- Prune users who haven't committed to the path in the last two years.
- Clamp list to 5 or less, as available.
- If list is empty, use all-time author list instead and mark as #old.

### CODE REVIEW

**Comment by rhalina** [comment]
Missing for neutron:
@rhalina 
@swagner-de 
@mutax 
@BenjaminLudwigSAP 
@m-kratochvil 

I can't find BenTs gh.com handle

**Comment by rhalina** [comment]
We need all escalation managers on-call in the * default set, as they need to be able to ACK in emergencies.

Fabian Krönner (handle unknown)
@artherd42 (Martin Vossen)
Ran Cliff (handle unknown)
@martinpink (Martin Pink)

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,297 @@
+*                                                      @BugRoger @databus23 @dhoeller @edda @geisslet @artherd42 @martinpink
+/.github/CODEOWNERS                                    @BugRoger @databus23 @dhoeller @edda @geisslet @artherd42 @martinpink
+
+/ci                                                    @auhlig @Kuckkuck @businessbean @jknipper @fwiesel
+
+/common/helm3-helper                                   @SchwarzM  # old
+/common/inventory-updater                              @BerndKue @stanislav-zaprudskiy
```
Comment (stanislav-zaprudskiy) [inline .github/CODEOWNERS:7]:
```suggestion
/common/inventory-updater                              @BerndKue
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+/common/manila-provisioner                             @jknipper  # old
+/common/mariadb                                        @galkindmitrii @stefanhipfel @Carthaca @IvoGoman @fwiesel
+/common/memcached                                      @auhlig @businessbean @majewsky @SuperSandro2000 @bashar-alkhateeb
+/common/mysql_metrics                                  @galkindmitrii @Carthaca
+/common/nats                                           @notandy
+/common/owner-info                                     @majewsky @SuperSandro2000
+/common/pgbackup                                       @majewsky @SuperSandro2000 @ArtieReus @VoigtS
+/common/pgmetrics                                      @majewsky @SuperSandro2000 @VoigtS
+/common/postgresql                                     @majewsky @fwiesel @auhlig @VoigtS @SuperSandro2000
+/common/postgresql-ng                                  @SuperSandro2000 @majewsky @stanislav-zaprudskiy @jknipper @stefanhipfel
```
Comment (stanislav-zaprudskiy) [inline .github/CODEOWNERS:18]:
```suggestion
/common/postgresql-ng                                  @SuperSandro2000 @majewsky @jknipper @stefanhipfel
```

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+/prometheus-exporters/cloudprober                      @artherd42 @defo89
+/prometheus-exporters/cronus-exporter                  @dhalimi @majewsky @LimorGargir @richardtief @geisslet
+/prometheus-exporters/cronus-openstack-alerts          @dhalimi
+/prometheus-exporters/cronus-reporter                  @dhalimi
+/prometheus-exporters/cronus-simulator                 @dhalimi @majewsky
+/prometheus-exporters/cronus-updater                   @dhalimi @majewsky
+/prometheus-exporters/documentation-prober             @dhalimi @majewsky  # old
+/prometheus-exporters/event-exporter                   @jknipper @databus23 @Nuckal777
+/prometheus-exporters/hermes-query-exporter            @Kuckkuck
+/prometheus-exporters/ipmi-exporter                    @stefanhipfel @BerndKue @stanislav-zaprudskiy
```
Comment (stanislav-zaprudskiy) [inline .github/CODEOWNERS:142]:
```suggestion
/prometheus-exporters/ipmi-exporter                    @stefanhipfel @BerndKue
```

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+/openstack/tempest/octavia-tempest                     @misamoylov @fwiesel @majewsky
+/openstack/tempest/tempest-base                        @misamoylov @fwiesel @stefanhipfel
+/openstack/trust-manager                               @fwiesel
+/openstack/utils                                       @fwiesel @notandy @galkindmitrii @Carthaca @joker-at-work
+/openstack/vcenter-operator                            @joker-at-work @fwiesel @majewsky @databus23
+/openstack/vcf                                         @chuan137
+/openstack/volume-claims                               @majewsky @fwiesel
+
+/prometheus-exporters/apic-exporter                    @artherd42 @IvoGoman @geisslet
+/prometheus-exporters/arista-exporter                  @BerndKue @stanislav-zaprudskiy @stefanhipfel
```
Comment (stanislav-zaprudskiy) [inline .github/CODEOWNERS:131]:
```suggestion
/prometheus-exporters/arista-exporter                  @BerndKue @stefanhipfel
```

**Comment by mbrowny** [comment]
@grandchild
CODEOWNERS: Add CoreDNS colleagues to /opentsack/designate and /openstack/designate-nanny (@mbrowny @vssldmtrv @SoenkeL)

**Comment by m-kratochvil** [comment]
According to the CODEOWNERS file check, following users need to work out their write access to the repo:
@VoigtS
@andypf
@dhalimi
@christopherhans

**Comment by hemna** [comment]
possibly missing openstack/os-brick and openstack/oslo* ?

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+/openstack/keppel-trivy                                @majewsky @SuperSandro2000 @VoigtS @Nuckal777
+/openstack/keystone                                    @galkindmitrii @Carthaca @bbobrov @fwiesel @rajivmucheli
+/openstack/kos-operator                                @stefanhipfel @databus23
+/openstack/kubernetes                                  @fwiesel
+/openstack/kubernikus                                  @fwiesel @jknipper
+/openstack/limes                                       @majewsky @SuperSandro2000 @VoigtS @Nuckal777
+/openstack/lyra                                        @databus23 @ArtieReus
+/openstack/maia                                        @artherd42 @viennaa @richardtief @Kuckkuck @dhalimi1975
+/openstack/manila                                      @Carthaca @chuan137 @galkindmitrii @kpawar-sap
+/openstack/manila-antelope                             @Carthaca @dusandordevicsap
```
Comment (Carthaca) [inline .github/CODEOWNERS:95]:
```suggestion
/openstack/manila-antelope                             @Carthaca @chuan137 @galkindmitrii @kpawar-sap
```

File: .github/CODEOWNERS [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+/openstack/manila-nanny                                @chuan137 @Carthaca @dusandordevicsap
```
Comment (Carthaca) [inline .github/CODEOWNERS:96]:
```suggestion
/openstack/manila-nanny                                @Carthaca @chuan137 @galkindmitrii @kpawar-sap
```

**Review by Carthaca — COMMENTED** [review]
Maybe the 3 manila rules even can be combined into one, but I'm not entirely sure about the syntax, i.e. I don't know if `/openstack/manila*` would do what I want: cover all files in the charts nested below, so I went for suggesting explicit changes instead

**Comment by grandchild** [comment]
> possibly missing openstack/os-brick and openstack/oslo* ?

These paths don't exist in this repo.

File: .github/CODEOWNERS [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+/prometheus-exporters/cronus-openstack-alerts          @dhalimi1975
+/prometheus-exporters/cronus-reporter                  @dhalimi1975
+/prometheus-exporters/cronus-simulator                 @dhalimi1975
+/prometheus-exporters/cronus-updater                   @dhalimi1975
+/prometheus-exporters/documentation-prober             @dhalimi1975  # old
+/prometheus-exporters/event-exporter                   @jknipper @databus23
+/prometheus-exporters/hermes-query-exporter            @Kuckkuck @notque
+/prometheus-exporters/ipmi-exporter                    @stefanhipfel @BerndKue
+/prometheus-exporters/ipmi-service-discovery           @auhlig  # old
+/prometheus-exporters/ironic-exporter                  @BerndKue @stefanhipfel
```
Comment (sandzwerg) [inline .github/CODEOWNERS:142]:
```suggestion
/prometheus-exporters/ironic-exporter                  @BerndKue @stefanhipfel @sandzwerg 

```

So we have at least three

File: .github/CODEOWNERS [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+/prometheus-exporters/nfs-exporter                     @auhlig  # old
+/prometheus-exporters/ns-exporter                      @databus23
+/prometheus-exporters/ntp-exporter                     @majewsky @SuperSandro2000 @VoigtS @Nuckal777
+/prometheus-exporters/octobus-query-exporter           @max-len @IvoGoman @Kuckkuck @christopherhans @kevin-fischer
+/prometheus-exporters/oomkill-exporter                 @jknipper @defo89
+/prometheus-exporters/opensearch-logs-query-exporter   @Kuckkuck @dimtass @chuan137 @databus23
+/prometheus-exporters/openstack-exporter               @viennaa @hemna @crenduchinta88 @kevin-fischer
+/prometheus-exporters/ping-exporter                    @auhlig @defo89  # old
+/prometheus-exporters/poller-simulator                 @dhalimi1975 @corey-aloia
+/prometheus-exporters/redfish-exporter                 @BerndKue @stefanhipfel
```
Comment (sandzwerg) [inline .github/CODEOWNERS:159]:
```suggestion
/prometheus-exporters/redfish-exporter                 @BerndKue @stefanhipfel @sandzwerg 
```

File: .github/CODEOWNERS [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+/prometheus-exporters/cloudprober                      @artherd42 @defo89
+/prometheus-exporters/cronus-exporter                  @dhalimi1975 @LimorGargir @richardtief @geisslet
+/prometheus-exporters/cronus-openstack-alerts          @dhalimi1975
+/prometheus-exporters/cronus-reporter                  @dhalimi1975
+/prometheus-exporters/cronus-simulator                 @dhalimi1975
+/prometheus-exporters/cronus-updater                   @dhalimi1975
+/prometheus-exporters/documentation-prober             @dhalimi1975  # old
+/prometheus-exporters/event-exporter                   @jknipper @databus23
+/prometheus-exporters/hermes-query-exporter            @Kuckkuck @notque
+/prometheus-exporters/ipmi-exporter                    @stefanhipfel @BerndKue
```
Comment (sandzwerg) [inline .github/CODEOWNERS:140]:
```suggestion
/prometheus-exporters/ipmi-exporter                    @stefanhipfel @BerndKue @sandzwerg 
```

**Review by sandzwerg — COMMENTED** [review]
Added myself to some ironic related path, so we have at least three people for these

File: .github/CODEOWNERS [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+/openstack/keppel                                      @majewsky @SuperSandro2000 @VoigtS @Nuckal777
+/openstack/keppel-trivy                                @majewsky @SuperSandro2000 @VoigtS @Nuckal777
+/openstack/keystone                                    @galkindmitrii @Carthaca @bbobrov @fwiesel @rajivmucheli
+/openstack/kos-operator                                @stefanhipfel @databus23
+/openstack/kubernetes                                  @fwiesel
+/openstack/kubernikus                                  @fwiesel @jknipper
+/openstack/limes                                       @majewsky @SuperSandro2000 @VoigtS @Nuckal777
+/openstack/lyra                                        @databus23 @ArtieReus
+/openstack/maia                                        @notque @artherd42 @viennaa @richardtief @Kuckkuck @dhalimi1975
+/openstack/manila*                                     @Carthaca @chuan137 @galkindmitrii @kpawar-sap
```
Comment (Carthaca) [inline .github/CODEOWNERS:94]:
```suggestion
/openstack/manila*                                     @Carthaca @chuan137 @galkindmitrii @kpawar-sap @sumitarora2786 @crenduchinta88
```

Adds Sumit and Chandra from Storage Team to Manila CodeOwners

File: .github/CODEOWNERS [hunk 11]
```diff
@@ -0,0 +1,295 @@
...
+/openstack/tempest/cinder-tempest                      @misamoylov @fwiesel
+/openstack/tempest/designate-tempest                   @misamoylov @fwiesel
+/openstack/tempest/glance-tempest                      @misamoylov @fwiesel @rajivmucheli
+/openstack/tempest/ironic-tempest                      @misamoylov @fwiesel
+/openstack/tempest/keystone-tempest                    @misamoylov @fwiesel @rajivmucheli
+/openstack/tempest/manila-tempest                      @kpawar-sap @misamoylov @fwiesel
+/openstack/tempest/neutron-tempest                     @misamoylov @fwiesel
+/openstack/tempest/nova-tempest                        @misamoylov @fwiesel
+/openstack/tempest/octavia-tempest                     @misamoylov @fwiesel
+/openstack/tempest/tempest-base                        @misamoylov @fwiesel @stefanhipfel
```
Comment (galkindmitrii) [inline .github/CODEOWNERS:121]: could you please add me for all tempest* paths?

Comment (grandchild) [reply .github/CODEOWNERS:121]: only `.../tempest*` or all `.../*tempest*`?

Comment (grandchild) [reply .github/CODEOWNERS:121]: eh, probably the latter.

Comment (galkindmitrii) [reply .github/CODEOWNERS]: all /*tempest* please :)

**Comment by businessbean** [comment]
https://github.com/sapcc/helm-charts/pull/6299/commits/677fec09048ae5d47f0dbf70e225df15a135d282 done because of https://github.com/sapcc/helm-charts/pull/6380

File: .github/CODEOWNERS [hunk 12]
```diff
@@ -0,0 +1,296 @@
...
+/system/kubernikus-metal                               @databus23 @defo89 @Nuckal777 @jknipper @xsen84
+/system/kubernikus-monitoring                          @uwe-mayer @databus23 @jknipper @Nuckal777
+/system/kubernikus-rbac                                @databus23
+/system/ldap-named-user                                @defo89
+/system/logs                                           @Kuckkuck @IvoGoman @businessbean @corey-aloia @timojohlo @notque
+/system/maintenance-controller                         @Nuckal777 @defo89 @majewsky @SuperSandro2000 @VoigtS
+/system/manila-csi-plugin                              @jknipper
+/system/metallb                                        @defo89  # old
+/system/metis                                          @IvoGoman @richardtief
+/system/metis-api                                      @IvoGoman
```
Comment (IvoGoman) [inline .github/CODEOWNERS:254]:
```suggestion
/system/metis                                          @IvoGoman @timojohlo @richardtief
/system/metis-api                                      @IvoGoman @timojohlo
```

File: .github/CODEOWNERS [hunk 13]
```diff
@@ -0,0 +1,296 @@
...
+/openstack/tempest/nova-tempest                        @misamoylov @fwiesel @galkindmitrii
+/openstack/tempest/octavia-tempest                     @misamoylov @fwiesel @galkindmitrii
+/openstack/tempest/tempest-base                        @misamoylov @fwiesel @stefanhipfel @galkindmitrii
+/openstack/trust-manager                               @fwiesel
+/openstack/utils                                       @fwiesel @notandy @galkindmitrii @Carthaca @joker-at-work
+/openstack/vcenter-operator                            @joker-at-work @fwiesel @databus23
+/openstack/vcf                                         @chuan137
+/openstack/volume-claims                               @fwiesel
+
+/prometheus-exporters/apic-exporter                    @artherd42 @IvoGoman @geisslet
```
Comment (IvoGoman) [inline .github/CODEOWNERS:128]:
```suggestion
/prometheus-exporters/apic-exporter                    @artherd42 @IvoGoman
```

**Pull request merged**